service/test/_iothub_registry_manager_test.js

```diff
@@ -0,0 +1,689 @@
+/* eslint-disable no-var */
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+var assert = require('chai').assert;
+var sinon = require('sinon');
+var IoTHubRegistryManager = require('../dist/cl/iothub_registry_manager').IoTHubRegistryManager;
+
+var testCredentials = {
+  signRequest: sinon.stub().callsFake(function (webResource) {
+    return Promise.resolve(webResource);
+  }),
+  getHubName: sinon.stub().returns('fake.host.name')
+};
+
+describe('IoTHubRegistryManager', function () {
+  it(`Constructor creates an instance of the IoTHubRegistryManager`, function (testCallback) {
+    var ioTHubRegistryManager = new IoTHubRegistryManager(testCredentials);
+    assert.instanceOf(ioTHubRegistryManager, IoTHubRegistryManager);
+    testCallback();
+  });
+
+  it('createDeviceWithSas calls the createOrUpdateIdentity method on the PL client', async () => {
+    var returnValue = {
+        interfaces: {
+          testInterfaceInstanceName: {}
+        }
+      };
+    var testDeviceId = 'testDeviceId';
+    var testPrimaryKey = 'testPrimaryKey';
```

Comment:
>'testPrimaryKey' [](start = 25, length = 16)

I fear we will get dinged for hard coded keys in the code.  I know I have.  Substitute with a uuid?

Reply:
for the unit tests I don't think we need other than "a" value